olp-cpp-sdk-dataservice-read/tests/ConfigApiTest.cpp

```diff
@@ -0,0 +1,147 @@
+  }
+
+ private:
+  DoCallApiCallback call_api_callback_;
+};
+
+const std::string kCatalog = "hrn:here:data::some_catalog";
+const std::string kURI = "/catalogs/" + kCatalog;
+const std::string kBillingTag = "billingTag";
+
+TEST(ConfigApiTest, GetCatalog) {
+  OlpClient client;
+  CancellationContext context;
+
+  {
+    SCOPED_TRACE("Test input HTTP params without billing tag");
+
+    ConfigApiTestable config_api(
+        [](std::string path, std::string method,
+           std::multimap<std::string, std::string> query_params,
+           std::multimap<std::string, std::string> header_params,
+           std::multimap<std::string, std::string> form_params,
+           std::shared_ptr<std::vector<unsigned char>> post_body,
+           std::string content_type, CancellationContext context) {
+
+          EXPECT_EQ(path, kURI);
+          EXPECT_EQ(method, "GET");
+          EXPECT_TRUE(query_params.empty());
+          EXPECT_EQ(query_params.find(kBillingTag), query_params.end());
+
+          EXPECT_TRUE(post_body == nullptr);
+          EXPECT_TRUE(content_type.empty());
+
+          return HttpResponse(olp::http::HttpStatusCode::OK);
+        });
+
+    auto response =
+        config_api.GetCatalog(client, kCatalog, boost::none, context);
+    EXPECT_TRUE(response.IsSuccessful());
+  }
+
+  {
+    SCOPED_TRACE("Test valid billing tag");
+
+    const std::string kTag = "test_tag";
+
+    ConfigApiTestable config_api(
+        [&](std::string path, std::string method,
+            std::multimap<std::string, std::string> query_params,
+            std::multimap<std::string, std::string> header_params,
+            std::multimap<std::string, std::string> form_params,
+            std::shared_ptr<std::vector<unsigned char>> post_body,
+            std::string content_type, CancellationContext context) {
+
+          auto billing_tag_iter = query_params.find(kBillingTag);
+          EXPECT_NE(billing_tag_iter, query_params.end());
+          if (billing_tag_iter != query_params.end()) {
+            EXPECT_EQ(billing_tag_iter->second, kTag);
+          }
+
+          return HttpResponse(olp::http::HttpStatusCode::OK);
+        });
+
+    auto response = config_api.GetCatalog(client, kCatalog, kTag, context);
+    EXPECT_TRUE(response.IsSuccessful());
+  }
+
+  {
+    SCOPED_TRACE("Test failed response");
+
+    const std::string kErrorMessage = "test_error";
+    const auto kHttpStatusCode = olp::http::HttpStatusCode::BAD_REQUEST;
+
+    ConfigApiTestable config_api(
+        [&](std::string path, std::string method,
+            std::multimap<std::string, std::string> query_params,
+            std::multimap<std::string, std::string> header_params,
+            std::multimap<std::string, std::string> form_params,
+            std::shared_ptr<std::vector<unsigned char>> post_body,
+            std::string content_type, CancellationContext context) {
+          return HttpResponse(kHttpStatusCode, kErrorMessage);
+        });
```

Comment:
With mocked method this could be changed to:
```cpp
EXPECT_CALL(config_api, DoCallApi)
    .WillOnce(Return(HttpResponse(kHttpStatusCode, kErrorMessage)));
```

Reply:
Not relevant anymore